packages/core/src/fhir-deduplication/shared.ts

```
@@ -49,72 +50,90 @@ function createExtensionRelatedArtifact(resourceType: string, id: string | undef
+  const originalExtension = "extension" in target ? [...target.extension] : [];
+  mutativeDeepMerge(target, source, keepExtensions);
 
   // This part combines resources together and adds the ID references of the duplicates into the master resource
   // regardless of whether new information was found
 
-  if (!isExtensionIncluded) {
-    delete combined.extension;
-  } else if ("extension" in r1) {
-    combined.extension = [...r1.extension, extensionRef];
+  if (!keepExtensions) {
+    delete target.extension;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)